test/unit/SourceLocator/Type/DirectorySourceLocatorTest.php

```diff
@@ -0,0 +1,136 @@
+        $classes = $reflector->getAllClasses();
+        $this->assertCount(2, $classes);
+        $classNames = [];
+        foreach ($classes as $clazz) {
+            $classNames[] = $clazz->getName();
+        }
+        sort($classNames);
+        $this->assertEquals('BetterReflectionTest\Assets\DirectoryScannerAssets\Bar\FooBar', $classNames[0]);
+        $this->assertEquals('BetterReflectionTest\Assets\DirectoryScannerAssets\Foo', $classNames[1]);
+    }
+
+    public function testScanDirectoryFiles()
+    {
+        $class = new \ReflectionClass(get_class($this->sourceLocator));
+        $method = $class->getMethod('scan');
+        $method->setAccessible(true);
+        $result = $method->invokeArgs($this->sourceLocator, [$this->directoryToScan]);
+        $this->assertCount(3, $result);
+
+        // test file path
+        $files = [];
+        foreach ($result as $file) {
+            $clazz = new \ReflectionClass('BetterReflection\SourceLocator\Type\SingleFileSourceLocator');
+            $property = $clazz->getProperty('filename');
+            $property->setAccessible(true);
+            $files[] = realpath($property->getValue($file));
+        }
+        sort($files);
+        $this->assertEquals(realpath(__DIR__ . '/../../Assets/DirectoryScannerAssets/Bar/Empty.php'), $files[0]);
+        $this->assertEquals(realpath(__DIR__ . '/../../Assets/DirectoryScannerAssets/Bar/FooBar.php'), $files[1]);
+        $this->assertEquals(realpath(__DIR__ . '/../../Assets/DirectoryScannerAssets/Foo.php'), $files[2]);
+
+        // test class names
+        $classNames = [];
+        foreach ($result as $file) {
+            /* @var $file SingleFileSourceLocator */
+            $reflector = new ClassReflector($file);
+            foreach ($reflector->getAllClasses() as $clazz) {
+                $classNames[] = $clazz->getName();
+            }
+        }
+        sort($classNames);
+        $this->assertEquals('BetterReflectionTest\Assets\DirectoryScannerAssets\Bar\FooBar', $classNames[0]);
+        $this->assertEquals('BetterReflectionTest\Assets\DirectoryScannerAssets\Foo', $classNames[1]);
+        $this->assertCount(2, $classNames);
+    }
+
+    public function testInvalidDirectory()
+    {
+        $this->expectException(InvalidDirectory::class);
+        new DirectorySourceLocator([substr($this->directoryToScan, 0, strlen($this->directoryToScan)-1)]);
+    }
+
+    public function testInvalidIntegerDirectory()
+    {
+        $this->expectException(InvalidDirectory::class);
+        new DirectorySourceLocator([$this->directoryToScan, 1]);
+    }
+
+    public function testInvalidBoleanDirectory()
+    {
+        $this->expectException(InvalidDirectory::class);
+        new DirectorySourceLocator([$this->directoryToScan, true]);
+    }
+
+    public function testInvalidObjectDirectory()
+    {
+        $this->expectException(InvalidDirectory::class);
+        new DirectorySourceLocator([$this->directoryToScan, new \stdClass()]);
```

Comment:
Remember to test combos like `[$validPath, $invalidPath]`, to verify that you are checking all values